bettercli/cl/selector.py

```
@@ -146,31 +147,32 @@ def decorator(options:'list[str]'):
 
         sys.stdout.write(ANSIColors.reset)
         self.print()
-        keyboard.on_press(self.handler)
+        keyboard.on_press(self.handler, suppress=True)
         validated = False
         while not validated:
             while not self.ENTERED:
-                # Add a small sleep to prevent CPU spinning
-                time.sleep(0.1)
+                pass
             if (validate := self._validator(self.selected)) != True:
```

Ruff (0.8.2)

Avoid inequality comparisons to `True`; use `if not validate := self._validator(self.selected):` for false checks

Replace with `not validate := self._validator(self.selected)`

(E712)